c++/configure.ac

```diff
@@ -195,8 +200,63 @@ AS_IF([test "$with_openssl" != no], [
 ])
 AM_CONDITIONAL([BUILD_KJ_TLS], [test "$with_openssl" != no])
 
-# CapnProtoConfig.cmake.in needs this variable.
-AC_SUBST(WITH_OPENSSL, $with_openssl)
+# Fibers need the symbols getcontext, setcontext, swapcontext and makecontext
+AS_IF([test "$with_fibers" != no], [
+  libc_supports_fibers=yes
+  AC_SEARCH_LIBS([makecontext], [], [], [
+    libc_supports_fibers=no
+  ])
+  AC_SEARCH_LIBS([getcontext], [], [], [
+    libc_supports_fibers=no
+  ])
+  AC_SEARCH_LIBS([setcontext], [], [], [
+    libc_supports_fibers=no
+  ])
+  AC_SEARCH_LIBS([swapcontext], [], [], [
+    libc_supports_fibers=no
+  ])
+
+  AS_IF([test "$libc_supports_fibers" = yes], [
+      with_fibers=yes
+    ], [
+    # If getcontext does not exist in libc, try with libucontext
+    ucontext_supports_fibers=yes
+    AC_CHECK_LIB(ucontext, [makecontext], [], [
+      ucontext_supports_fibers=no
+    ])
+    AC_CHECK_LIB(ucontext, [getcontext], [], [
+      ucontext_supports_fibers=no
+    ])
+    AC_CHECK_LIB(ucontext, [setcontext], [], [
+      ucontext_supports_fibers=no
+    ])
+    AC_CHECK_LIB(ucontext, [swapcontext], [], [
+      ucontext_supports_fibers=no
+    ])
+    AS_IF([test "$ucontext_supports_fibers" = yes], [
+      ASYNC_LIBS="$ASYNC_LIBS -lucontext"
+      with_fibers=yes
+    ], [
+      AS_IF([test "$with_fibers" = yes], [
+        AC_MSG_ERROR([Missing symbols required for fibers (makecontext, setcontext, ...). Disable fibers (--without-fibers) or install libucontext])
+      ], [
+        AC_MSG_WARN([could not find required symbols (makecontext, setcontext, ...) -- won't build with fibers])
+        with_fibers=no
+      ])
+    ])
+  ])
+])
+AS_IF([test "$with_fibers" = yes], [
+  CXXFLAGS="$CXXFLAGS -DKJ_USE_FIBERS"
+], [
+  CXXFLAGS="$CXXFLAGS -DKJ_USE_FIBERS=0"
+])
+
+# CapnProtoConfig.cmake.in needs these variables,
+# we force them to NO because we don't need the CMake dependency for them,
+# the dependencies are provided by the .pc files.
+AC_SUBST(WITH_OPENSSL, NO)
+AC_SUBST(_WITH_LIBUCONTEXT, NO)
```

Comment:
Out of curiosity, why does one of these have a leading underscore and the other doesn't?

Reply:
I think (@Sarcasm should confirm) that it is to say that `WITH_OPENSSL` is in the public API (that's a user-facing option), and `_WITH_LIBUCONTEXT` is meant for internal use (and therefore should not be set by the user).

Reply:
I confirm, but feel free to drop the leading underscore if you think it looks better.